geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexImpl.java

```diff
@@ -226,6 +226,10 @@ public void destroy(boolean initiator) {
       getDataRegion().getExtensionPoint().removeExtension(extensionToDelete);
     }
 
+    // Remove cache service profile
+    dataRegion
```

Comment:
Should we do it as the last step of destroy 
 - to prevent any create index command to be successful (with the same name) while the old index was in the process of destroying.
 - checkCompatibility can prevent this if the cacheServiceProfile is still present in the region